src/ui/menu/Menu.tsx

```diff
@@ -1,15 +1,38 @@
 import React, { useMemo } from 'react';
-import { FixedSizeList } from 'react-window';
+import { FixedSizeList, ListChildComponentProps } from 'react-window';
 
-import { Menu as MuiMenu, makeStyles } from '@material-ui/core';
+import { Menu as MuiMenu, makeStyles, PopoverOrigin } from '@material-ui/core';
 
 import { Palette } from 'styles';
 
-import { MenuProps, MuiMenuStyleProps } from '.';
+namespace Menu {
+  type BaseProps<T> = {
+    id: string;
+    anchorEl: Element;
+    items: T[];
+    children: React.ComponentType<ListChildComponentProps>;
+    onClose(): void;
+    anchorOrigin?: PopoverOrigin;
+    transformOrigin?: PopoverOrigin;
+    width?: number;
+    height?: number;
+    itemSize?: number;
+  };
+
+  export type Props<T, R> = BaseProps<T> & R;
+
```

Comment:
export type Props<T, R> = BaseProps<T> & R;
What does it mean?
It will copy BaseProps and extend Props with R object props?

Reply:
It means we have BaseProps type with generic parameter. I can add as single type definition but it will looks ugly. We need to merge BaseProps with generic R parameter because MenuItem can have generic component. We need to define type definition for this component so instead of one big type with & R i created proxy type which makes this prettier